patches/tModLoader/Terraria/Projectile.cs.patch

```diff
@@ -414,16 +414,66 @@
  			if (wet)
  				position += wetVelocity;
  			else
-@@ -13080,6 +_,9 @@
+@@ -13006,6 +_,7 @@
+ 
+ 			fisher.playerFishingConditions = Main.player[owner].GetFishingConditions();
+ 			int baitItemType = fisher.playerFishingConditions.BaitItemType;
++
+ 			if (baitItemType == 2673) {
+ 				Main.player[owner].displayedFishingInfo = Language.GetTextValue("GameUI.FishingWarning");
+ 				if ((fisher.X < 380 || fisher.X > Main.maxTilesX - 380) && fisher.waterTilesCount > 1000 && !NPC.AnyNPCs(370)) {
+@@ -13080,12 +_,23 @@
  			FishingCheck_RollEnemySpawns(ref fisher);
  			FishingCheck_RollItemDrop(ref fisher);
  			bool flag = false;
-+			PlayerHooks.CatchFish(Main.player[owner], Main.player[owner].inventory[Main.player[owner].selectedItem],
-+				fisher.waterNeededToFish, fisher.inLava ? 1 : fisher.inHoney ? 2 : 0, fisher.fishingLevel, fisher.heightLevel, fisher.questFish, ref fisher.rolledItemDrop);
++			string sonarText = null;
++			Color color = Color.White;
++			PlayerHooks.CatchFish(Main.player[owner], fisher, ref sonarText, ref color);
 +
  			if (fisher.rolledItemDrop > 0) {
- 				if (Main.player[owner].sonarPotion) {
- 					Item item = new Item();
++				if (Main.player[owner].sonarPotion && (sonarText == null || sonarText.Trim().Length > 0)) {
++
+-				if (Main.player[owner].sonarPotion) {
++					if (sonarText == null) {
+-					Item item = new Item();
++						Item item = new Item();
+-					item.SetDefaults(fisher.rolledItemDrop);
++						item.SetDefaults(fisher.rolledItemDrop);
```

Comment:
These indentation changes in patches are not welcome, figure out some code change that won't require these

Reply:
Gotos are good for this.